lib/commands/timeout.js

```diff
@@ -29,22 +31,22 @@ helpers.startNewCommandTimeout = function (cmd) {
   // if command timeout is 0, it is disabled
   if(!this.newCommandTimeoutMs) return;
 
-  //  TODO: replace setTimeout by CancellableDelay
-  this.noCommandTimer = setTimeout(() => {
-    let onCommand = async () => {
+  this.noCommandTimer = util.cancellableDelay(this.newCommandTimeoutMs);
+  this.noCommandTimer
+    .then(async () => {
       log.warn(`Shutting down because we waited ` +
                `${this.newCommandTimeoutMs / 1000} seconds for a command`);
       await this.startUnexpectedShutdown(new Error(`Command ${cmd} timed out!`));
-    };
-    B.resolve(onCommand()).done();
-  }, this.newCommandTimeoutMs);
+    })
+    .catch(B.CancellationError, (/*err*/) => {
+      // ignore
+    });
 };
 
-helpers.implicitWaitForCondition = async function (condFn) {
+helpers.implicitWaitForCondition = async (condFn) => {
```

Comment:
I'm curious how this works. We needed to not use arrow functions before since they bind the function when we want to bind it when we add these helper functions to the prototype.